msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/IBroker.java

```diff
@@ -67,11 +67,17 @@ default IAuthenticationResult parseBrokerAuthResult(String authority, String idT
             if (idToken != null) {
                 builder.idToken(idToken);
                 if (accountId != null) {
-                    String idTokenJson =
-                            JWTParser.parse(idToken).getParsedParts()[1].decodeToString();
+                    String idTokenJson;
+
+                    try {
+                        idTokenJson = new String(Base64.getDecoder().decode(idToken.split("\\.")[1]), StandardCharsets.UTF_8);
```

Comment:
jwt's are base64url encoded, not base64. THis already results in a P1. 

Can we add some tests around it?

Reply:
This was mistakenly copied from a branch that had not yet been updated with the fix for the previous issue. 

In the latest commit the logic both here and in `TokenRequestExecutor` has been moved to a helper method in `JsonHelper`, and new tests were added to confirm the behavior.